spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/web/embedded/TomcatWebServerFactoryCustomizerTests.java

```diff
@@ -194,6 +197,14 @@ void customMaxHttpRequestHeaderSize() {
 			.isEqualTo(DataSize.ofMegabytes(10).toBytes()));
 	}
 
+	@ParameterizedTest
+	@ValueSource(ints = { -1, 0, 100 })
```

Comment:
Is `@ParameterizedTest` required here to test negative value? @wilkinsona

Reply:
I don't think so. Given that all three values are passed through and then returned by `getMaxParameterCount()` without any having special treatment, I think testing with a single value would be sufficient.

Reply:
OK, I reverted this part.